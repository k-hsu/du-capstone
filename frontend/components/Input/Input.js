import React, { forwardRef } from "react";
import styled from "@emotion/styled";
import Text from "../Text/Text";
import { color, fontWeight, spacing } from "../../theme";

const StyledFlex = styled.label`
  display: flex;
  flex-direction: column;
`;

const StyledInput = styled.input`
  border-radius: ${spacing["0.5"]};
  padding: ${spacing["0.5"]} ${spacing["0.75"]};
  font-family: "IBM Plex Sans", sans-serif;
`;

const Input = forwardRef(
  ({ id, labelText, errorMessage, autoFocus, ...props }, ref) => {
    return (
      <StyledFlex htmlFor={id}>
        <Text fontWeight={fontWeight.bold}>{labelText}</Text>
        <StyledInput id={id} ref={ref} autoFocus={autoFocus} {...props} />
        {errorMessage && <Text color={color.red}>{errorMessage}</Text>}
      </StyledFlex>
    );
  }
);
Input.displayName = "Input";

export default Input;
