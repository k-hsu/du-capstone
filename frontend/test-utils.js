import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { renderHook } from "@testing-library/react-hooks";

function setup(jsx) {
  return {
    user: userEvent.setup(),
    ...render(jsx),
  };
}

export * from "@testing-library/react";
export { renderHook, setup };
