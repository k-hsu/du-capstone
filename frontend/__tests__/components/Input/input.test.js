import React, { useRef } from "react";
import { render, screen, userEvent } from "../../../test-utils";
import Input from "../../../components/Input/Input";

describe("Input", () => {
  const RefComponent = (props) => {
    const ref = useRef();
    return <Input ref={ref} {...props} />;
  };
  it("should render input component and its associated label", () => {
    render(<RefComponent labelText="band" />);
    expect(screen.getByText("band")).toBeInTheDocument();
    expect(screen.getByRole("textbox", { name: "band" })).toBeInTheDocument();
  });
  it("should show values as typed by user", () => {
    render(<RefComponent labelText="band" />);

    userEvent.type(
      screen.getByRole("textbox", { name: "band" }),
      "Eye of the Tiger"
    );
    expect(screen.getByRole("textbox", { name: "band" })).toHaveValue(
      "Eye of the Tiger"
    );
  });
});
