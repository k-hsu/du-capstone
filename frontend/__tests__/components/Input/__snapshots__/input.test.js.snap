// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render input component styles 1`] = `
<label
  className="css-1iyoj2o"
>
  <p
    className="css-z3zo0q"
    fontFamily="IBM Plex Sans"
    fontWeight="700"
  >
    prompt
  </p>
  <input
    className="css-5jib8k"
  />
</label>
`;
